src/plugins/intel_cpu/src/emitters/snippets/x64/jit_loop_emitters.cpp

```diff
@@ -52,7 +53,8 @@ void jit_loop_begin_emitter::emit_code(const std::vector<size_t> &in, const std:
 
 void jit_loop_begin_emitter::emit_impl(const std::vector<size_t>& in, const std::vector<size_t>& out) const {
     // If the loop evaulate once, we can skip loop begin code emission
-    if (evaluate_once)
+    // If work_amount is dynamic, we should get runtime `work_amount` - it might be `zero` and we should skip loop evaluation
+    if (evaluate_once && !is_work_amount_dynamic)
```

Comment:
Good question!
Loops will be always executed only once (or not more than once :smile:) if `work_amount` is dynamic and `evaluate_once == true`. The same for static loops. We don't have any checks for this case in dynamic and static scenarios. We set them only in the corresponding passes. So I'm not sure that we can throw exception somewhere even (because it's like `make action A` and immediately `check that action A is completed`)
BTW, If `work_amount` is dynamic, we should put the runtime value to register anyway to skip loop evaluation if needed.